cronjobs/FileLoader.py

```diff
@@ -0,0 +1,128 @@
+
+from models import ScheduleHistory, CallingPointHistory
+from darwinpush.messagefactories.xml import ScheduleXMLMessageFactory
+import darwinpush.xb.pushport as pp
+
+import re
+
+class ScheduleFileLoader:
+
+    def __init__(self, file_location):
+        assert(file_location)
+        self.file_location = file_location
+
+    def update_daily_schedules(self):
+
+        collect_mode=False
+        journey_buffer = []
+
+        for line in open(self.file_location):
+
+            if '<Journey' in line:
+                assert(not collect_mode)
+                collect_mode=True
+
+            elif '</Journey' in line:
+                assert(collect_mode)
+
+                journey_buffer.append(line)
+                journey_string = "".join(journey_buffer)
+                journey_string = journey_string.replace("Journey", "schedule")
+                journey_string = journey_string.replace("<OR", "<ns2:OR")
+                journey_string = journey_string.replace("<PP", "<ns2:PP")
+                journey_string = journey_string.replace("<IP", "<ns2:IP")
+                journey_string = journey_string.replace("<DT", "<ns2:DT")
+                journey_string = journey_string.replace("<OPOR", "<ns2:OPOR")
+                journey_string = journey_string.replace("<OPIP", "<ns2:OPIP")
+                journey_string = journey_string.replace("<OPDT", "<ns2:OPDT")
+                journey_string = journey_string.replace("<OPPP", "<ns2:OPPP")
+                journey_string = journey_string.replace("cancelReason", "ns2:cancelReason")
+                journey_string = re.sub(r'plat=\".+\"', '', journey_string)
+                journey_string = re.sub(r'qtrain=\".+\"', '', journey_string)
+                journey_string = re.sub(r'can=\".+\"', '', journey_string)
+
+                journey_string = (  '<?xml version="1.0" encoding="UTF-8"?>'
+                                    + '<Pport ts="2015-07-20T11:52:07.3487919+01:00" version="12.0" xmlns="http://www.thalesgroup.com/rtti/PushPort/v12" xmlns:ns2="http://www.thalesgroup.com/rtti/PushPort/Schedules/v1">'
+                                    + '<uR requestID="0000000000020608" requestSource="at09" updateOrigin="CIS">'
+                                    + journey_string
+                                    + '</uR>'
+                                    + '</Pport>')
+
+                self.add_schedule_from_buffer(journey_string)
+
+                journey_buffer = []
+                collect_mode=False
+
+            if collect_mode:
+                journey_buffer.append(line)
+
+    def add_schedule_from_buffer(self, journey_buffer):
+        r = pp.CreateFromDocument(journey_buffer)
+        assert(None is r.sR)
+        assert(None is not r.uR)
+        assert(1 == len(r.uR.schedule))
+
+        m = ScheduleXMLMessageFactory.build(r.uR.schedule[0], r, journey_buffer)
+
+        # We try to find a schedule, and replace it if we do
+        found = (ScheduleHistory
```

Comment:
I thought the whole point of having this history was not to do this update.

Am I getting something wrong?